index.js

```diff
@@ -228,7 +228,12 @@ function fromString (that, string, encoding) {
   var length = byteLength(string, encoding) | 0
   that = createBuffer(that, length)
 
-  that.write(string, encoding)
+  var actual = that.write(string, encoding)
+
+  if (actual !== length) {
```

Comment:
This happens when writing a hex string that contains invalid characters. For example, writing `abcxxabc` will only write `abc`. Node ignores everything after the first invalid character.

You can test it:

```js
Buffer.from('abcdxxabcd', 'hex').toString('hex')
```